dubbo-remoting/dubbo-remoting-http3/src/main/java/org/apache/dubbo/remoting/transport/netty4/NettyHttp3Server.java

```diff
@@ -18,104 +18,76 @@
     }
 
     @Override
     protected void doOpen() throws Throwable {
         bootstrap = new Bootstrap();
-
         bossGroup = NettyEventLoopFactory.eventLoopGroup(1, EVENT_LOOP_BOSS_POOL_NAME);
-
         NettyServerHandler nettyServerHandler = new NettyServerHandler(getUrl(), this);
         channels = nettyServerHandler.getChannels();
-
-        FrameworkModel frameworkModel = ScopeModelUtil.getFrameworkModel(getUrl().getScopeModel());
-        NettyHttp3ProtocolSelectorHandler selectorHandler =
-                new NettyHttp3ProtocolSelectorHandler(getUrl(), frameworkModel);
-
-        int idleTimeout = UrlUtils.getIdleTimeout(getUrl());
-        io.netty.channel.ChannelHandler codec = Helper.configCodec(Http3.newQuicServerCodecBuilder(), getUrl())
-                .sslContext(Http3SslContexts.buildServerSslContext(getUrl()))
-                .maxIdleTimeout(idleTimeout, MILLISECONDS)
-                .tokenHandler(InsecureQuicTokenHandler.INSTANCE)
-                .handler(new ChannelInitializer<QuicChannel>() {
-                    @Override
-                    protected void initChannel(QuicChannel ch) {
-                        ch.pipeline()
-                                .addLast(nettyServerHandler)
-                                .addLast(new IdleStateHandler(0, 0, idleTimeout, MILLISECONDS))
-                                .addLast(new Http3ServerConnectionHandler(new ChannelInitializer<QuicStreamChannel>() {
-                                    @Override
-                                    protected void initChannel(QuicStreamChannel ch) {
-                                        ch.pipeline()
-                                                .addLast(NettyHttp3FrameCodec.INSTANCE)
-                                                .addLast(new HttpWriteQueueHandler())
-                                                .addLast(selectorHandler);
-                                    }
-                                }));
-                    }
-                })
-                .build();
-
-        // bind
         try {
             ChannelFuture channelFuture = bootstrap
                     .group(bossGroup)
                     .channel(NioDatagramChannel.class)
-                    .handler(codec)
+                    .handler(Http3Helper.configCodec(Http3.newQuicServerCodecBuilder(), getUrl())
+                            .sslContext(Http3SslContexts.buildServerSslContext(getUrl()))
+                            .tokenHandler(InsecureQuicTokenHandler.INSTANCE)
+                            .handler(new ChannelInitializer<QuicChannel>() {
+                                @Override
+                                protected void initChannel(QuicChannel ch) {
+                                    ChannelPipeline pipeline = ch.pipeline();
+                                    pipelineConfigurator.accept(pipeline);
```

Comment:
Please ensure that it is not null for safety.

Reply:
Checking in constructor.